Style/Frontend/index.md

```diff
@@ -0,0 +1,1451 @@
+
+    ...
+
+  }
+  ```
+
+## Testing
+
+- New components should always be accompanied by a corresponding jest test. Learn about [jest](https://jestjs.io/) and read through our tests to get an idea of how we utilize it
+- Bug fixes should always be accompanied by a corresponding regression test
+- Be cautious about stubs and mocks - they can make your tests more brittle
+- Strive to write many small pure functions, and minimize where mutations occur
+- 100% test coverage is a good goal to strive for, even if it’s not always practical to reach it
+
+## Comments & Docstrings
+
+- Make use of `TODO`s, `FIXME`s and `NOTE`s where it makes sense
+
+  ```jsx
+  // TODO for things that need to be done
+  // FIXME for things that don't work as they should
+  // NOTE for implementation details, edge cases, etc
+
+  const calculateEPS = () => {
+    // TODO: finish implementing eps calculation
+    return 0;
+  }
+
+  // FIXME: not returning expected values
+  const square(x) => x * 2;
+
+  // NOTE: this component expects xyz to be present in order to work properly
+  const ComplicatedComponent = ({props}) => { ... } 
+  ```
+
+- If you comment something out because you don't need it, delete it instead
+
+  ```jsx
+  // bad
+  const styles = StyleService.create({
+    featureContainer: {
+      flex: 1,
+      // justifyContent: 'space-between'
+    }
+  });
+
+  // an exception
+  // it is acceptable to comment code that will be uncommented in the future
+  // just make sure to note it with a FIXME or TODO
+  const styles = StyleService.create
+    // TODO: Style for a currently disabled component that needs some work
+    // featureContainer: {
+    //  flex: 1,
+    // }
+  });
+  ```
+
+- New components and screens should always have a good docstring with a general description, list of params and what they are, and any important info you want to communicate to other devs who will come across it
+
+  ```jsx
+  /**
+   * Widget component for showing data with a particular color and style
+   * @param data {Object} data object to be rendered, should be in xyz format otherwise it will not render properly
+   * @param [color='highlight'] {String} optional color string to color data with
+   * @param [style={}] {Style} optional style to apply to container
+   */
+  const Widget = ({data, color = 'highlight', style = {}}) => { ... }
+  ```
+
+## Imports & Exports
+
+- Use modules
+
+  ```jsx
+  // bad
+  const MyLib = require('@/path/MyLib');
+  module.exports = MyLib.doThing;
+  
+  // getting there
+  import MyLib from '@/path/MyLib';
+  export default MyLib.doThing;
+  
+  // best
+  import {MyLib} from '@/path/MyLib';
+  export default doThing;
+  ```
+
+
+- Use `@` imports over relative imports
+
+  ```jsx
+  // bad
+  import {MyLib} from './MyLib';
+  
+  // good
+  import {MyLib} from '@/path/MyLib';
+  ```
+
+- Avoid wildcard imports
+
+  ```jsx
+  // bad
+  import * as MyLib from '@/path/MyLib';
+  
+  // good
+  import MyLib from '@/path/MyLib';
```

Comment:
The good version would be extracting the exact exported funcs you need. Cause if there is no default export, this wouldn't work

Reply:
I added a little thing that explains that you should do this because it forces yourself to add that default export, github wont show it for some reason